plugins/pipelines/jobBadge.js

```diff
@@ -47,13 +45,15 @@ module.exports = config => ({
             const { jobFactory } = request.server.app;
             const { pipelineFactory } = request.server.app;
             const { id, jobName } = request.params;
-            const badgeService = request.server.app.ecosystem.badges;
-            const { encodeBadgeSubject } = request.server.plugins.pipelines;
             const { statusColor } = config;
             const badgeConfig = {
-                badgeService,
-                statusColor,
-                encodeBadgeSubject
+                statusColor
+            };
+
+            const getBadge = badgeObject => {
```

Comment:
move this function out of the handler function

Reply:
Updated